### ISSUE TITLE
rename pm2ps_dict() to System()

### ISSUE THREAD

**Issue by claytonpbarrows** [NREL/PowerSystems.jl#433]
[This function name](https://github.com/NREL/PowerSystems.jl/blob/master/src/parsers/pm2ps_parser.jl#L15) is misleading, it should probably be `System()` or `pm2system()` or something like that...

**Comment by daniel-thom** [comment]
Just `System` with the current function parameters is very generic (`Dict{String, Any)}`.  It could collide with a future function that also takes a dict.  We could create a type for the input data and pass that.

Example:
```
abstract type PowerSystemsInputDataSource end
struct PowerModelsData <: PowerSystemsInputDataSource end

function System(::PowerModelsData, data::Dict{String, Any}; kwargs...)
```

I'm also OK with `pm2System`.

**Comment by jd-lara** [comment]
closed via #435
